src/torrent_info.cpp

```diff
@@ -528,7 +528,7 @@ namespace libtorrent
 		}
 		else
 		{
-			file_flags &= ~file_storage::flag_symlink;
+			file_flags &= ~(file_storage::flag_symlink & 0xffffffff);
```

Comment:
yes, that's the problem. I wanted to make the minimal amount of changes, to make `file_flags_t` unsigned, I need to change `file_storage::file_flags` too.  I will give it a try